ds3-autogen-c/src/main/java/com/spectralogic/ds3autogen/c/CCodeGenerator.java

```diff
@@ -72,6 +72,7 @@ private void generateCommands() throws IOException, TemplateException {
         if (null != spec.getTypes()) {
```

Comment:
In the utils package there is a class called ConverterUtils that has isEmpty and hasContent functions.  I suggest using these when you want to check if lists, sets, maps, and strings are empty/null vs full.